esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -603,27 +605,37 @@ bool EspHttpBinding::basicAuth(IEspContext* ctx)
     if(user == NULL)
     {
         WARNLOG("Can't find user in context");
+        ctx->setAuthError(EspAuthErrorUserNotFoundInContext);
         ctx->AuditMessage(AUDIT_TYPE_ACCESS_FAILURE, "Authentication", "Access Denied: No username provided");
         return false;
     }
 
     if(m_secmgr.get() == NULL)
     {
         WARNLOG("No mechanism established for authentication");
+        ctx->setAuthError(EspAuthErrorNoAuthMechanism);
         return false;
     }
 
     ISecResourceList* rlist = ctx->queryResources();
     if(rlist == NULL)
+    {
+        WARNLOG("No Security Resource");
+        ctx->setAuthError(EspAuthErrorEmptySecResource);
         return false;
+    }
 
     bool authenticated = m_secmgr->authorize(*user, rlist, ctx->querySecureContext());
     if(!authenticated)
     {
+        const char *desc = nullptr;
         if (user->getAuthenticateStatus() == AS_PASSWORD_EXPIRED || user->getAuthenticateStatus() == AS_PASSWORD_VALID_BUT_EXPIRED)
-            ctx->AuditMessage(AUDIT_TYPE_ACCESS_FAILURE, "Authentication", "ESP password is expired");
+            desc = "ESP password is expired";
```

Comment:
Should be able to call ctx->setRespMsg() and no need for "desc" pointer

Reply:
The "desc" is also used by the AuditMessage().